src/edu/cscc/javaadventure/engine/combat/CharacterCombatant.java

```diff
@@ -0,0 +1,167 @@
+package edu.cscc.javaadventure.engine.combat;
+
+import edu.cscc.javaadventure.Character;
+import edu.cscc.javaadventure.Sword;
+import edu.cscc.javaadventure.Weapon;
+import edu.cscc.javaadventure.engine.InvalidRollException;
+
+public class CharacterCombatant implements Combatant{
+    private Character character;
+    private Weapon currentWeapon;
+    private Combatant combatant;
+    private int initiative;
+    private int baseArmorClassBonus;
+
+    //Initializing with the current default weapon
+
+    public CharacterCombatant (Character character) {
+        this.character= character;
+        this.currentWeapon= new Sword();
+        this.baseArmorClassBonus = 10;
+    }
+
+    public void setInitiative(int initiative) {
+        this.initiative = initiative;
+    }
+
+    @Override
+    public void receiveDamage(int damage) {
+
+        character.setHealth(character.getHealth() - damage);
+        //int damageReceived = combatant.getHealth() - damage;
+    }
+
+    @Override
+    public boolean isActiveCombatant() {
+        return character.getHealth() > 0;
+    }
+
+    @Override
+    public String getName() {
+        return this.character.getName();
+    }
+
+    @Override
+    public Weapon getWeapon() {
+        return currentWeapon;
+    }
+
+    @Override
+    public int getInitiative() {
+        return this.combatant.getInitiative();
+    }
+
+    @Override
+    public Integer getArmorClass() {
+        return this.baseArmorClassBonus + calculateDexterityBonus();
+    }
+
+    @Override
+    public Integer getHealth() {
+        return this.combatant.getHealth();
+    }
+
+
+    @Override
+    public void equipWeapon(Weapon weapon) {
+        try {
+            if(weapon != null) {
+                if(weapon.getTwoHanded() && getStrength() < 15) {
+                    throw new UnableToWieldWeaponException("Character does not have enough weapon");
+                }
+                currentWeapon = weapon;
+            }
+        } catch (UnableToWieldWeaponException e) {
+            System.err.println("Unable to equip weapon: " + e.getMessage());
+        }
+    }
+
+
+    @Override
+    public String attack(int roll, Combatant target) {
```

Comment:
Does not need a try/catch. Method signature should throw OpponentIncapacitatedException and InvalidRollException